### PR TITLE
calico-host-protection: add psp rule

### PR DESCRIPTION
**Pull request by knrt10**
Calico host endpoint controller only needs to talk to the apiserver and
should not be granted privileged PSP.

Update deprecated apiVersion for clusterrole and clusterrolebinding.

closes: #287
Signed-off-by: knrt10 <kautilya@kinvolk.io>

### CODE REVIEW

**Comment by knrt10** [comment]
`kc auth can-i --list --as="system:serviceaccount:kube-system:calico-hostendpoint-controller"` has both 

```
podsecuritypolicies.policy                      []                  [zz-minimal]      [use]
podsecuritypolicies.policy                      []                  [zz-privileged]   [use]
```

listed as we have set privileged policy for all `Serviceaccounts` in `kube-system` namespace. 

Pod created using `calico-hostendpoint-controller`, still does not use `zz-minimal` policy. 

Trying to debug why this is happening

File: assets/charts/control-plane/calico-host-protection/templates/psp.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+metadata:
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
+  name: calico-hostendpoint-controller-psp
+spec:
+  privileged: false
+  fsGroup:
+    rule: RunAsAny
+  runAsUser:
+    rule: RunAsAny
```
Comment (invidian) [inline assets/charts/control-plane/calico-host-protection/templates/psp.yaml:12]: Why it is allowed to run as any if we run it as non-root?

File: assets/charts/control-plane/calico-host-protection/templates/psp.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
```
Comment (invidian) [inline assets/charts/control-plane/calico-host-protection/templates/psp.yaml:2]: I think we could make this policy more strict. Start with restricted policy and then allow what's required.

File: assets/charts/control-plane/calico-host-protection/templates/host-endpoint-controller.yaml [hunk 3]
```diff
@@ -43,7 +48,7 @@ spec:
 
 ---
 # rbac.yaml
-apiVersion: rbac.authorization.k8s.io/v1beta1
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (invidian) [inline assets/charts/control-plane/calico-host-protection/templates/host-endpoint-controller.yaml:51]: Version bump could be done in separate commit.

File: assets/charts/control-plane/calico-host-protection/templates/host-endpoint-controller.yaml [hunk 4]
```diff
@@ -33,6 +33,11 @@ spec:
       containers:
       - image: quay.io/kinvolk/calico-hostendpoint-controller:v0.0.4
         name: calico-hostendpoint-controller
+        securityContext:
+          privileged: false
```
Comment (invidian) [inline assets/charts/control-plane/calico-host-protection/templates/host-endpoint-controller.yaml:37]: Are all those statements required? Shouldn't just `runAsUser: 65534` be sufficient to make it not run as root?

File: test/platform/packet/packet_test.go [hunk 5]
```diff
@@ -98,3 +98,22 @@ func TestWhenBGPIsNotDisabledInConfigurationServersHasBGPSessionCreated(t *testi
 		t.Fatalf("Worker pool with BGP not disabled should have at least one BGP session")
 	}
 }
+
+func TestAnnotationForCalicoHostEndpointController(t *testing.T) {
```
Comment (invidian) [inline test/platform/packet/packet_test.go:102]:
```suggestion
func TestCalicoHostEndpointControllerRunsWithDedicatedPSP(t *testing.T) {
```

**Review by invidian — CHANGES_REQUESTED** [review]

File: assets/charts/control-plane/calico-host-protection/templates/psp.yaml (hunk 1 above)
Comment (knrt10) [reply assets/charts/control-plane/calico-host-protection/templates/psp.yaml:12]: Fixed

**Review by invidian — APPROVED** [review]

**Review by surajssd — APPROVED** [review]
LGTM

**Pull request merged**